cli/test/unit/tasks/install_spec.js

```diff
@@ -45,6 +54,12 @@ describe('/lib/tasks/install', function () {
 
       // sinon.stub(os, 'tmpdir').returns('/tmp')
       sinon.stub(util, 'isCi').returns(false)
+
+      // on CircleCI we have "normal" terminal, but without stdout.columns
+      // so work around it and let it use progress bar
+      const isTerm = util.hasTerminalDimensions()
+      sinon.stub(util, 'hasTerminalDimensions').returns(isTerm || isCircle)
```

Comment:
I don't understand why this is necessary. Shouldn't we stub it out both ways? Why make it dynamic?

It seems like the logic of this is pretty simple - it just boils down to this: if we're in CI or we aren't attached to a TTY then use the verbose logger - else use the progress logger. Is it any more complicated than that?

Also - what I don't understand is isn't that what its doing right now? When you posted the pictures it seems that it *was* using the verbose renderer and not the progress. What then is this doing that's different?

Reply:
the posted picture is AFTER updating the logic and taking "process.stdout.columns" into account! The twisted logic in the test is to simulate Docker output on CircleCI where we actually have the full terminal working correctly (I think this is because they allow you to SSH into running build later)